ethernet_init/ethernet_init.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * SPDX-FileCopyrightText: 2022 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Unlicense OR CC0-1.0
+ */
+#pragma once
+
+#include "esp_eth_driver.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+typedef enum {
+    ETH_DEV_TYPE_UNKNOWN,
+    ETH_DEV_TYPE_INTERNAL_ETH,
+    ETH_DEV_TYPE_SPI,
+} eth_dev_type;
```

Comment:
by convention in IDF, typedefs end with `_t`
```suggestion
} eth_dev_type_t;
```